src/cephadm/cephadmlib/systemd_unit.py

```diff
@@ -2,22 +2,138 @@
 
 import contextlib
 import os
+import json
 import pathlib
 
-from typing import IO, List, Optional, Union
+from typing import Any, Dict, IO, List, Optional, Union
 
 from . import templating
 from .call_wrappers import call_throws
 from .container_engines import Docker, Podman
 from .context import CephadmContext
 from .daemon_identity import DaemonIdentity, DaemonSubIdentity
+from .exceptions import SystemdDropInException
 from .file_utils import write_new
 from .logging import write_cluster_logrotate_config
 
 
 _DROP_IN_FILENAME = '99-cephadm.conf'
 
 
+class SystemdDropIn:
```

Comment:
Do you remember when I wrote this* class and it was declined? :-)

*OK not exactly the same, but similar.